com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/security/AuthorizationChecker.java

```diff
@@ -46,6 +49,19 @@ public AuthorizationChecker(CloudControllerClientProvider clientProvider, SpaceG
         this.clientProvider = clientProvider;
         this.spaceGetter = spaceGetter;
         this.applicationConfiguration = applicationConfiguration;
+        if (spaceDevelopersCache == null) {
+            initSpaceDevelopersCache();
+        }
+    }
+
+    private void initSpaceDevelopersCache() {
+        synchronized (AuthorizationChecker.class) {

```

Comment:
The class is a Spring singleton, so why not make the cache non-static. That way you'll be able to just make this method "private synchronized void".

Reply:
Fixed